src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+        c.argument('rg_location', help='Location of this resource group. Leave empty if the resource group location will be specified during the blueprint assignment.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+        c.argument('display_name', help='Display name of this resource group artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('tags', tags_type, help='Tags to be assigned to this resource group.')
+
+    with self.argument_context('blueprint resource-group update') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('rg_name', help='Name of this resource group. Leave empty if the resource group name will be specified during the blueprint assignment.')
+        c.argument('rg_location', help='Location of this resource group. Leave empty if the resource group location will be specified during the blueprint assignment.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+        c.argument('display_name', help='Display name of this resource group artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('tags', tags_type, arg_group='Resource Group', help='Tags to be assigned to this resource group.')
+
+    with self.argument_context('blueprint resource-group delete') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+
+    with self.argument_context('blueprint resource-group show') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+
+    with self.argument_context('blueprint resource-group list') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint artifact policy create') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('policy_definition_id', help='Azure resource ID of the policy definition.')

```

Comment:
As we discussed before, _id is kept for guid. And for policy definition parameter, is it possible to support both resource id and name?

Reply:
I think the help message from swagger is wrong. An example value is "/providers/Microsoft.Authorization/policySetDefinitions/1f3afdf9-d0c9-4c3d-847f-89da613e70a8", the corresponding ARM resource id is "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/providers/Microsoft.Authorization/policyAssignments/SecurityCenterBuiltIn". 

There's existing [code](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/custom.py#L843) in azure-cli to use name or id for both policy set definition and the resource, but it needs to use the policy SDK. As this command is beyond the cutline of breadth coverage, we may not want to spend extra effort to go that far.

I'll name the option to be 'policy-definition', modify help message and apply similar changes to 'role-definition-id'.